glotaran/io/register.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import os
+
+import xarray as xr
+
+from glotaran.model import Model
+from glotaran.parameter import ParameterGroup
+from glotaran.project import Result
+from glotaran.project import SavingOptions
+from glotaran.project import Scheme
+
+from .io import Io
+
+_io_register = {}
+
+
+def register_io(fmt: str | list[str], io: Io):
+    fmt = fmt if isinstance(fmt, list) else [fmt]
+    for fmt_name in fmt:
+        _io_register[fmt_name] = io()
```

Comment:
This should give a warning if `fmt_name in _io_register`, e.g. a user has two plugins that register the same format.

Reply:
Btw we have the same problem in `glotaran/model/register.py` with `register_model`.